core/src/solve.jl

```diff
@@ -79,11 +82,8 @@ struct Basin{C} <: AbstractParameterNode
     drainage::Vector{Float64}
     infiltration::Vector{Float64}
     # cache this to avoid recomputation
-    current_level::Vector{Float64}
-    current_area::Vector{Float64}
-    # The derivative of the area with respect to the level
-    # used for the analytical Jacobian
-    current_darea::Vector{Float64}
+    current_level::Union{Vector{Float64}, DiffCache{Vector{Float64}}}
+    current_area::Union{Vector{Float64}, DiffCache{Vector{Float64}}}
```

Comment:
For these and other Unions, a type parameter wasn't working?

Reply:
Yes It is, I changed it to parameterized now. I'm not sure it solves the performance issue though.